ui/src/BulkUploads.js

```diff
@@ -0,0 +1,302 @@
+    formData.append("file", e.target.files[0]);
+
+    const fileName = e.target.files[0].name;
+    // Reset the file
+    e.target.value = null;
+
+    // Send the file data to the backend
+    axios
+      .request({
+        method: "post",
+        url: "/api/upload",
+        data: formData,
+        headers: {
+          "Content-Type": "multipart/form-data",
+        },
+        onUploadProgress: (p) => {
+          console.log(p);
+
+          // Update file upload progress
+          this.setState({
+            fileProgress: p.loaded / p.total,
+          });
+        },
+      })
+      .then((data) => {
+        // send the job details
+        const fileId = data.data;
+        const jobData = new FormData();
+        jobData.append("fileId", fileId);
+        jobData.append("fileName", fileName);
+        jobData.append("collectionExerciseId", this.props.collectionExerciseId);
+        // TODO: Temp hardcoded to BULK_REFUSAL
+        jobData.append("jobType", "BULK_REFUSAL");
+
+        const response = fetch(`/api/job`, {
+          method: "POST",
+          body: jobData,
+        });
+
+        if (!response.ok) {
+          // TODO - nice error handling
+          // If we check it, it's is currently buggy and leaves the popup on the screen for unknown reasons - need to raise a defect
+        }
+
+        // Hide the progress dialog and flash the snackbar message
+        this.setState({
+          fileProgress: 1.0,
+          fileUploadSuccess: true,
+          uploadInProgress: false,
+        });
+
+        this.getJobs();
+      });
+  };
+
+  handleClose = (event, reason) => {
+    // Ignore clickaways so that the dialog is modal
+    if (reason === "clickaway") {
+      return;
+    }
+
+    this.setState({
+      fileUploadSuccess: false,
+    });
+  };
+
+  getJobs = async () => {
+    // TODO: Job Type Temp hardcoded to BULD_REFUSAL
+    const response = await fetch(
+      `/api/job?collectionExercise=${this.props.collectionExerciseId}&jobType=BULK_REFUSAL`
+    );
+
+    // TODO: We need more elegant error handling throughout the whole application, but this will at least protect temporarily
+    if (!response.ok) {
+      return;
+    }
+
+    const jobsJson = await response.json();
+
+    this.setState({ jobs: jobsJson });
+  };
+
+  handleOpenDetails = (job) => {
+    this.setState({ showDetails: true, selectedJob: job.id });
+  };
+
+  handleClosedDetails = () => {
+    this.setState({ showDetails: false });
+  };
+
+  onProcessJob = () => {
+    fetch(`/api/job/${this.state.selectedJob}/process`, {
+      method: "POST",
+    });
+  };
+
+  onCancelJob = () => {
+    fetch(`/api/job/${this.state.selectedJob}/cancel`, {
+      method: "POST",
+    });
+  };
+
+  render() {
+    const selectedJob = this.state.jobs.find(
+      (job) => job.id === this.state.selectedJob
+    );
+
+    const bulkRefusalTableRows = this.state.jobs.map(
+      (job, index) =>
+        this.state.authorisedActivities.includes(
```

Comment:
no need to check permissions here... there should be no data because we wouldn't call the backend if we didn't have permission